app/src/auth/utils/loginInterceptor.js

```diff
@@ -92,23 +89,23 @@ export function canHandleLoadError(err) {
 }
 
 export async function handleLoadError(err) {
-    if (!err.response) { throw err } // unexpected error
-    // server issues
-    if (err.response.status >= 500) {
+    if (err instanceof InvalidHexStringError) {
```

Comment:
Not sure how useful the separate error for the hex string is, we could omit it and just throw `ResourceNotFoundError`. But this is also ok like this.

Reply:
It's more about semantics, I feel, logic maybe too. `getValidId` doesn't seem to be the right place to throw a `ResourceNotFoundError`. I'll keep it unless someone stops me! 😉

Reply:
Sure, though I think it would be better semantically to catch this error in the product loading code and then bubble a resource error that is caught here. But that leads to wrapping it around every place where `getValidId` is used so I'm fine with this 🚀

Reply:
Ah right, I feel you. I don't like this special case \*here\* either.